fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java

```diff
@@ -523,6 +523,38 @@ private List<List<TScanRangeParams>> splitScanRangeParamByRowCount(List<TScanRan
         return result;
     }
 
+    private Map<Long, ComputeNode> getNodeMapInDefaultWarehouse() {
+        Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
+        Map<Long, ComputeNode> idToNode = new HashMap<>();
+        for (long id : warehouse.getAnyAvailableCluster().getComputeNodeIds()) {
+            idToNode.put(id, GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNode(id));
+        }
+        return idToNode;
+    }
+
+    private TNetworkAddress getExecHostPort(Reference<Long> nodeIdRef) {
+        // TODO: need to refactor after be split into cn + dn
+        // get node from default warehouse
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            Map<Long, ComputeNode> idToNode = getNodeMapInDefaultWarehouse();
+            return SimpleScheduler.getComputeNodeHost(ImmutableMap.copyOf(idToNode), nodeIdRef);
+        }
+        return SimpleScheduler.getBackendHost(this.idToBackend, nodeIdRef);
+    }
+
+    private TNetworkAddress getExecHostPortV2(long nodeId, List<TScanRangeLocation> locations,
+                                              Reference<Long> nodeIdRef) {
+        // TODO: need to refactor after be split into cn + dn
+        // get node from default warehouse
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            Map<Long, ComputeNode> idToNode = getNodeMapInDefaultWarehouse();
+            return SimpleScheduler.getHost(nodeId, locations, idToBackend,
```

Comment:
`idToBackend` is a subset of `idToNode`.

suggestions:
recover `SimpleScheduler.getHost` with one map parameter and update this function below.
```
if (runmode == shared data) {
    return SimpleScheduler.getHost(nodeId, locations, idToNode, nodeIdRef);
} else {
    return SimpleScheduler.getHost(nodeId, locations, idToBackend, nodeIdRef);
}
```